authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourcePropertyProvider.java

```diff
@@ -1,41 +1,8 @@
 package com.sequenceiq.authorization.service;
 
-import java.util.Collection;
-import java.util.EnumSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-
-import org.apache.commons.lang3.NotImplementedException;
-
 import com.sequenceiq.authorization.resource.AuthorizationResourceType;
-import com.sequenceiq.cloudbreak.auth.crn.Crn;
 
 public interface ResourcePropertyProvider {
```

Comment:
Yes the original interface had lot of methods but at least in one place. Yes interface segregation principle can be applied but now the developer need to be aware of all the different interfaces that can be used, complex interface hierarchies and I'm not sure the new solution is more understandable than the previous. I saw that you created nice documentation but I saw that people were not aware of such manual in the past as well. It might worth asking others as well about which one is easier to understand. 
cc: @sodre90, @lnardai, @bbihari, @lturcsanyi

Reply:
that why we have the enforce unit test :)